### PR TITLE
Multi channel

### PR DESCRIPTION
**Pull request by Kevin2**
add multi channel option to simulations
tests with pixel shuffle layers to upscale prediction
tests with hardtanh to restrict phase predictions
separate architectures for amp and phase training

### CODE REVIEW

File: radionets/dl_framework/architectures/res_exp.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
     GeneralELU,
 )
 from functools import partial
+from math import pi
 
 
 class SRResNet_small(nn.Module):
```
Comment (FeGeyer) [inline radionets/dl_framework/architectures/res_exp.py:13]: Maybe rename now to something different than `small`?

File: radionets/dl_framework/architectures/res_exp.py [hunk 2]
```diff
@@ -17,29 +18,58 @@ def __init__(self):
...
-            nn.Conv2d(64, 64, 3, stride=1, padding=1), nn.BatchNorm2d(64)
+            nn.Conv2d(64, 64, 3, stride=1, padding=1, bias=False), nn.BatchNorm2d(64)
         )
 
-        self.final = nn.Sequential(nn.Conv2d(64, 2, 9, stride=1, padding=4, groups=2),)
+        self.shuffle = nn.Sequential(
+            nn.Conv2d(64, 252, 3, stride=1, padding=1, bias=True),
+            nn.PixelShuffle(3),
+            nn.PReLU(),
+            # nn.Conv2d(28, 252, 3, stride=1, padding=1, bias=True),
```
Comment (FeGeyer) [inline radionets/dl_framework/architectures/res_exp.py:47]: Can this be deleted?

File: radionets/dl_framework/architectures/res_exp.py [hunk 3]
```diff
@@ -17,29 +18,58 @@ def __init__(self):
...
 
         x = x + self.postBlock(self.blocks(x))
 
+        x = self.shuffle(x)
+
         x = self.final(x)
 
-        x0 = self.symmetry_amp(x[:, 0]).reshape(-1, 1, 63, 63)
-        x1 = self.symmetry_imag(x[:, 1]).reshape(-1, 1, 63, 63)
+        x0 = self.symmetry_amp(x[:, 0]).reshape(-1, 1, 189, 189)
```
Comment (FeGeyer) [inline radionets/dl_framework/architectures/res_exp.py:70]: Shape should be variable instead of hardcoded

File: radionets/dl_framework/data.py [hunk 4]
```diff
@@ -264,6 +264,9 @@ def load_data(data_path, mode, fourier=False, source_list=False):
         dataset containing x and y images
     """
     bundle_paths = get_bundles(data_path)
-    data = [path for path in bundle_paths if re.findall("samp_" + mode, path.name)]
+    data = np.sort(
+        [path for path in bundle_paths if re.findall("samp_" + mode, path.name)]
+    )
+    data = sorted(data, key=lambda f: int(''.join(filter(str.isdigit, str(f)))))
```
Comment (FeGeyer) [inline radionets/dl_framework/data.py:270]: What is the advantage of this?

**Review by FeGeyer — CHANGES_REQUESTED** [review]
We should briefly talk about the comments

File: radionets/dl_framework/data.py (hunk 4 above)
Comment (Kevin2) [reply radionets/dl_framework/data.py:270]: when comparing with the wsclean results for the paper the sorting was necessary

File: radionets/dl_framework/architectures/res_exp.py (hunk 3 above)
Comment (Kevin2) [reply radionets/dl_framework/architectures/res_exp.py:70]: true

File: radionets/dl_framework/architectures/res_exp.py (hunk 2 above)
Comment (Kevin2) [reply radionets/dl_framework/architectures/res_exp.py:47]: I will have a look

**Comment by Kevin2** [comment]
pushed some changes to include your suggestions

**Review by FeGeyer — APPROVED** [review]
Nice!
